src/Parsley/Inline.php

```diff
@@ -28,7 +28,15 @@ class Inline
 	public function __construct(DOMNode $node, array $marks = [])
 	{
 		$this->createMarkRules($marks);
-		$this->html = trim(static::parseNode($node, $this->marks) ?? '');
+
+		$html = static::parseNode($node, $this->marks) ?? '';
+
+		// only trim HTML if it doesn't consist of only spaces
+		if (strlen($html) !== substr_count($html, ' ')) {
```

Comment:
I think this would be safer as it also works with other kinds of whitespace:

```suggestion
		if (trim($html) !== '') {
```